tests/integration/security/file_mounted_certs/main_test.go

```diff
@@ -0,0 +1,100 @@
+// +build integ
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package filemountedcerts
+
+import (
+	"strings"
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+var (
+	inst istio.Instance
+)
+
+const (
+	ProxyMetadataJson = `
+	{
+      "JWT_POLICY": "third-party-jwt",
+      "PILOT_CERT_PROVIDER": "mycopki",
+      "FILE_MOUNTED_CERTS": "true",
+      "ISTIO_META_TLS_CLIENT_CERT_CHAIN": "/client-certs/cert.pem",
+      "ISTIO_META_TLS_CLIENT_KEY": "/client-certs/key.pem",
+      "ISTIO_META_TLS_CLIENT_ROOT_CERT": "/client-certs/ca.pem",
+      "ISTIO_META_TLS_SERVER_CERT_CHAIN": "/server-certs/cert.pem",
+      "ISTIO_META_TLS_SERVER_KEY": "/server-certs/key.pem",
+      "ISTIO_META_TLS_SERVER_ROOT_CERT": "/server-certs/ca.pem",
+      "ISTIO_META_PROXY_XDS_VIA_AGENT": "false",
+      "ISTIO_METAJSON_METRICS_INCLUSIONS": '{\"sidecar.istio.io/statsInclusionPrefixes\": \"access_log_file,cluster,cluster_manager,control_plane,http,http2,http_mixer_filter,listener,listener_manager,redis,runtime,server,stats,tcp,tcp_mixer_filter,tracing\"}',
+      "ISTIO_META_DNS_CAPTURE": "false",
+      "NIKO": "test2"
+	}
+`
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		Label(label.CustomSetup).
+
+		// SDS requires Kubernetes 1.13
+		RequireEnvironmentVersion("1.13").
+		Label("CustomSetup").
+		//Setup(istio.Setup(&inst, setupConfig, cert.CreateCustomEgressSecret)).
+		Setup(istio.Setup(&inst, setupConfig)).
+		Run()
+
+}
+
+
+func setupConfig(_ resource.Context, cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+
+	cfg.ControlPlaneValues = `
+components:
+  egressGateways:
+  - enabled: false
+    name: istio-egressgateway
+  ingressGateways:
+  - enabled: false
+    name: istio-ingressgateway
+  pilot:
+    enabled: true
+
+meshConfig:
+  defaultConfig:
+    controlPlaneAuthPolicy: "NONE"
+    discoveryAddress: istiod.istio-system:15010
+    proxyMetadata: ` + strings.Replace(ProxyMetadataJson, "\n", "", -1) +
+`
+values:
+  global:
+    logging:
+      level: "default:debug"
+    proxy:
+      logLevel: debug
+      componentLogLevel: "misc:debug"
+  pilot:
+    env:
+      PILOT_SKIP_VALIDATE_TRUST_DOMAIN: "true"
```

Comment:
We set a bunch of other env values for Pilot, ex: DNS_ADDR, ENABLE_CA_SERVER, PILOT_ENABLE_PROTOCOL_SNIFFING_*. Are they being set somewhere else?

Reply:
@shakti-das , I am actually not sure if those are set. will double-check. Thank you!

Reply:
The following env vars are present in the istiod pod:
```
    - name: REVISION
      value: default
    - name: JWT_POLICY
      value: first-party-jwt
    - name: PILOT_CERT_PROVIDER
      value: istiod
    - name: POD_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.name
    - name: POD_NAMESPACE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.namespace
    - name: SERVICE_ACCOUNT
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.serviceAccountName
    - name: KUBECONFIG
      value: /var/run/secrets/remote/config
    - name: PILOT_SKIP_VALIDATE_TRUST_DOMAIN
      value: "true"
    - name: PILOT_TRACE_SAMPLING
      value: "1"
    - name: PILOT_ENABLE_PROTOCOL_SNIFFING_FOR_OUTBOUND
      value: "true"
    - name: PILOT_ENABLE_PROTOCOL_SNIFFING_FOR_INBOUND
      value: "true"
    - name: INJECTION_WEBHOOK_CONFIG_NAME
      value: istio-sidecar-injector
    - name: ISTIOD_ADDR
      value: istiod.istio-system.svc:15012
    - name: PILOT_ENABLE_ANALYSIS
      value: "false"
    - name: CLUSTER_ID
      value: Kubernetes
    - name: CENTRAL_ISTIOD
      value: "false"
```

Reply:
We should set `ENABLE_CA_SERVER` as false (its default is true) and `DNS_ADDR` as "" (its default is ":15053").